src/common/mem.c

```diff
@@ -3231,6 +3232,10 @@ cache_evict_page(struct umem_cache *cache, bool for_sys)
 		/* The page is referenced by others while flushing */
 		if ((pinfo->pi_ref > 0) || is_page_dirty(pinfo) || pinfo->pi_io == 1)
 			return -DER_AGAIN;
+
+		/* The status of the page changed to non-evictable */
+		if (!is_id_evictable(cache, pinfo->pi_pg_id))
```

Comment:
It would be more clear if we check "pinfo->pi_evictable" here.

Reply:
fixed